vpr/src/route/route_timing.cpp

```diff
@@ -2007,35 +2006,66 @@ bool should_setup_lower_bound_connection_delays(int itry, const t_router_opts& /
 static bool is_better_quality_routing(const vtr::vector<ClusterNetId, t_traceback>& best_routing,
                                       const RoutingMetrics& best_routing_metrics,
                                       const WirelengthInfo& wirelength_info,
-                                      std::shared_ptr<const SetupHoldTimingInfo> timing_info) {
+                                      std::shared_ptr<const SetupHoldTimingInfo> timing_info,
+                                      const t_router_opts& router_opts) {
     if (best_routing.empty()) {
         return true; //First legal routing
     }
 
     //Rank first based on sWNS, followed by other timing metrics
     if (timing_info) {
-        if (timing_info->setup_worst_negative_slack() > best_routing_metrics.sWNS) {
-            return true;
-        } else if (timing_info->setup_worst_negative_slack() < best_routing_metrics.sWNS) {
-            return false;
-        }
+        if (router_opts.routing_budgets_algorithm != YOYO) {
+            // If RCV is disabled prioritize setup slack and don't worry about hold
 
-        if (timing_info->setup_total_negative_slack() > best_routing_metrics.sTNS) {
-            return true;
-        } else if (timing_info->setup_total_negative_slack() < best_routing_metrics.sTNS) {
-            return false;
-        }
+            if (timing_info->hold_worst_negative_slack() > best_routing_metrics.hWNS) {
+                return true;
+            } else if (timing_info->hold_worst_negative_slack() > best_routing_metrics.hWNS) {
+                return false;
+            }
 
-        if (timing_info->hold_worst_negative_slack() > best_routing_metrics.hWNS) {
-            return true;
-        } else if (timing_info->hold_worst_negative_slack() > best_routing_metrics.hWNS) {
-            return false;
-        }
+            if (timing_info->hold_total_negative_slack() > best_routing_metrics.hTNS) {
```

Comment:
Needs a comment as this looks like a repeat of the prior 5 lines of code until you look very closely (total negative slack instead of worst negative slack).